src/input/mod.rs

```diff
@@ -3070,31 +3070,50 @@ impl State {
 
         self.update_pointer_contents();
 
-        let scroll_factor = match source {
-            AxisSource::Wheel => self.niri.config.borrow().input.mouse.scroll_factor,
-            AxisSource::Finger => self.niri.config.borrow().input.touchpad.scroll_factor,
-            _ => None,
+        macro_rules! get_scroll_factors {
+            ($cfg:expr) => {{
+                let h_factor = $cfg
+                    .scroll_factor_horizontal
+                    .map(|x| x.0)
+                    .or_else(|| $cfg.scroll_factor.map(|x| x.0))
+                    .unwrap_or(1.);
+                let v_factor = $cfg
+                    .scroll_factor_vertical
+                    .map(|y| y.0)
+                    .or_else(|| $cfg.scroll_factor.map(|y| y.0))
+                    .unwrap_or(1.);
+                (h_factor, v_factor)
+            }};
+        }
```

Comment:
After you put the scroll-factor config into a `struct ScrollFactor`, this can be a proper method on that struct rather than a macro.

Reply:
Good call. I've since replaced the macro with `ScrollFactor::get_factors()` method, which is definitely cleaner and more maintainable.